### ISSUE TITLE
SSH stopped working

### ISSUE THREAD

**Issue by woolmonkey** [quexten/goldwarden#119]
So upgraded to 2.12 and ssh has stopped working.  I just get Permission denied (publickey).  It was working before the upgrade goldwarden ssh list shows all the values.  ssh-add -L says connection refused.  This was working on the previous version.

**Comment by quexten** [comment]
Thanks for the report, 0.2.12 changed a lot, I'll try to patch this ASAP.

>  ssh-add -L says connection refused. 

Please post the outputs of:
`goldwarden vault status`
`goldwarden config get-runtime-config`

**Comment by quexten** [comment]
Also does `goldwarden ssh list` show the key? (Don't post it, even though it's the public key)

**Comment by woolmonkey** [comment]
Goldwarden ssh list did show but in trying stuff I deleted .goldwarden.sock how do I get that recreated?  I tried reinstalling but nothing.
I was running AUR
/home/xxxx/.goldwarden-ssh-agent.sock      which is broken now because I removed the file.

**Comment by quexten** [comment]
> Goldwarden ssh list did show but in trying stuff I deleted .goldwarden.sock how do I get that recreated? I tried reinstalling but nothing.

systemctl --user restart goldwarden

>  /home/xxxxxx/.var/app/com.quexten.Goldwarden/data/ssh-auth-sock

Seems like a bug introduced recently. You can temporarily fix this by setting `GOLDWARDEN_SOCKET_PATH=/home/xxxx/.goldwarden.sock` in the systemd service file.

**Comment by quexten** [comment]
Actually `SSH_AUTH_SOCK=~/.goldwarden-ssh-agent.sock` for the ssh socket.

**Comment by quexten** [comment]
> systemctl --user restart goldwarden
If this does not work, run "goldwarden daemonize" and copy paste the output. It should show what's going on.

**Comment by woolmonkey** [comment]
Had a long post but much simpler.  12 well not create the socket, 10 well. When I run it with daemonize it errors out with this.
[]INF] [21:48] [Goldwarden > Keyring] >>> Creating new memguard keyring
[INF] [21:48] [Goldwarden > Agent] >>> Agent listening on ...
[INF] [21:48] [Goldwarden > SSH] >>> SSH Agent listening on 
Blocking, press ctrl+c to continue...
^X^Caccept error accept unix @abf34: use of closed network connection
removing sockets and exiting
unlinking 
accept error accept unix @abf34: use of closed network connection
accept error accept unix @abf34: use of closed network connection
accept error accept unix @abf34: use of closed network connection
accept error accept unix @abf34: use of closed network connection
accept error accept unix @abf34: use of closed network connection
accept error accept unix @abf34: use of closed network connection
no such file or directory
unlinking 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x5cc1bc42ff41]

goroutine 31 [running]:
github.com/quexten/goldwarden/agent.serveAgentSession({0x0, 0x0}, 0xc00016c0f0, 0xc0001b4000)
	github.com/quexten/goldwarden/agent/unixsocketagent.go:53 +0x61
created by github.com/quexten/goldwarden/agent.StartUnixAgent.func9 in goroutine 26
	github.com/quexten/goldwarden/agent/unixsocketagent.go:511 +0x49
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x5cc1bc42ff41]

goroutine 7 [running]:
github.com/quexten/goldwarden/agent.serveAgentSession({0x0, 0x0}, 0xc00016c0f0, 0xc0001b4000)
	github.com/quexten/goldwarden/agent/unixsocketagent.go:53 +0x61
created by github.com/quexten/goldwarden/agent.StartUnixAgent.func9 in goroutine 26
	github.com/quexten/goldwarden/agent/unixsocketagent.go:511 +0x49
(url)

**Comment by quexten** [comment]
Could you try https://github.com/quexten/goldwarden/pull/120? (To install just download the binary from the GitHub CI, chmod +x & rename it, replace /usr/bin/goldwarden and restart the daemon.

**Comment by woolmonkey** [comment]
I would but don't see how to download files there.

**Comment by quexten** [comment]
https://github.com/quexten/goldwarden/actions/runs/7952263026

Bottom of the page under artifacts

**Comment by woolmonkey** [comment]
Ahh had to be at the edge of the screen to scroll down.
No same error 
todd@Archie ~ goldwarden vault unlock
Error: dial unix: missing address
Is the daemon running?

Also noticed the ssh socket changed
SSH Agent listening on /home/todd/.ssh-agent-socket

So I changed my .bashrc to match but still get the top error.

And error of it running daemonized
```
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
accept error accept unix @fd064: use of closed network connection
removing sockets and exiting
unlinking 
no such file or directory
unlinking 
no such file or directory
memguard wiping memory and exiting

```

**Comment by quexten** [comment]
Could you try again on https://github.com/quexten/goldwarden/actions/runs/7984562192 ?

**Comment by LSeelig** [comment]
I am getting this too. Will do testing later.

**Comment by woolmonkey** [comment]
> Could you try again on https://github.com/quexten/goldwarden/actions/runs/7984562192 ?

Works except it moves the ssh socket
[Goldwarden > SSH] >>> SSH Agent listening on /home/todd/.ssh-agent-socket

Once I changed .bashrc it worked fine.

**Comment by quexten** [comment]
> Works except it moves the ssh socket

Where were you expecting the socket, and where did it get placed?

**Comment by woolmonkey** [comment]
It got placed in ~/.ssh-agent.socket

documentation mentions ~/.goldwarden-ssh-agent.sock

**Comment by quexten** [comment]
Ah, seems to be a bug introduced in 3b2d1fcf11f063653c44355e42529e8fea1d3888. Changed back to the path mentioned in the documentation in 966da90491f117428793ec9ea000dffcf676e91e

**Comment by quexten** [comment]
I *think* this is fixed is 0.2.13 now, if not, there is more logging now. Please re-open and post logs if it's not fixed.
